src/font.rs

```diff
@@ -0,0 +1,165 @@
+use crate::stream::Stream;
+use crate::Error;
+use std::fmt::Display;
+
+pub fn parse_font_families(text: &str) -> Result<Vec<FontFamily>, Error> {
+    let mut s = Stream::from(text);
+    let font_families = s.parse_font_families()?;
+
+    s.skip_spaces();
+    if !s.at_end() {
+        return Err(Error::UnexpectedData(s.calc_char_pos()));
+    }
+
+    Ok(font_families)
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub enum FontFamily {
+    Serif,
+    SansSerif,
+    Cursive,
+    Fantasy,
+    Monospace,
+    Named(String),
+}
+
+impl Display for FontFamily {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let str = match self {
+            FontFamily::Monospace => "monospace".to_string(),
+            FontFamily::Serif => "serif".to_string(),
+            FontFamily::SansSerif => "sans-serif".to_string(),
+            FontFamily::Cursive => "cursive".to_string(),
+            FontFamily::Fantasy => "fantasy".to_string(),
+            FontFamily::Named(s) => format!("\"{}\"", s),
+        };
+        write!(f, "{}", str)
+    }
+}
+
+impl<'a> Stream<'a> {
+    pub fn parse_font_families(&mut self) -> Result<Vec<FontFamily>, Error> {
+        let mut families = vec![];
+
+        while !self.at_end() {
+            self.skip_spaces();
+
+            let family = {
+                let ch = self.curr_byte()?;
+                if ch == b'\'' || ch == b'\"' {
+                    let res = self.parse_string()?;
+                    FontFamily::Named(res.to_string())
```

Comment:
Return `&str`?

Reply:
so I should have it take a Cow? because I also need to be able to pass an owned string (see below).

Reply:
Are you sure? You're never modifying the input string.

Reply:
Yeah, here:

```rs
let joined = idents.join(" ");

                    // TODO: No CSS keyword must be matched as a family name...
                    match joined.as_str() {
                        "serif" => FontFamily::Serif,
                        "sans-serif" => FontFamily::SansSerif,
                        "cursive" => FontFamily::Cursive,
                        "fantasy" => FontFamily::Fantasy,
                        "monospace" => FontFamily::Monospace,
                        _ => FontFamily::Named(joined),
                    }
```

The spec requires us to parse multiple spaces as a single space (which should again probably be done at the CSS parsing level, but since it's relatively straight-forward I just implemented it here).